wallet-connect/src/client.rs

```diff
@@ -274,29 +273,22 @@ impl Middleware for WCMiddleware<Provider<Client>> {
             .request("eth_signTransaction", vec![tx_obj])
             .await
             .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))?;
+
         let tx_rlp = rlp::Rlp::new(tx_bytes.as_ref());
-        // TODO: check that the decoded request matches the typed transaction content here?
-        let signature = match tx {
-            TypedTransaction::Eip1559(_) => {
-                let decoded_request = Eip1559TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
-                    .map(|x| x.1)
-                    .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))
-            }
-            TypedTransaction::Eip2930(_) => {
-                let decoded_request = Eip2930TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
-                    .map(|x| x.1)
-                    .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))
-            }
-            TypedTransaction::Legacy(_) => {
-                let decoded_request = TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
+        let signature = match tx_rlp.data() {
```

Comment:
good to keep the TODO note because there may be another bug 😁 
```suggestion
        // TODO: check that the decoded request matches the typed transaction content here? or a new `sign_transaction` function that returns both request+signature?
        let signature = match tx_rlp.data() {
```